### ISSUE TITLE
Linux link error: undefined reference to `std::vector<bs::ShaderVariation::Param, bs::StdAlloc<bs::ShaderVariation::Param, bs::GenAlloc> >::~vector()'

### ISSUE THREAD

**Issue by procedural** [BearishSun/BansheeEngine#178]
```bash
[699/707] Linking CXX executable /home/constantine/Engines/BansheeEngine/bin/x64/Release/ExampleLowLevelRendering
FAILED: /home/constantine/Engines/BansheeEngine/bin/x64/Release/ExampleLowLevelRendering 
: && /usr/bin/clang++  -Wall -fPIC -fno-exceptions -fno-strict-aliasing -fno-rtti -fno-ms-compatibility -Wl,-rpath= -g0 -O2 -DNDEBUG -Wno-unused-variable  -s -no-pie Examples/ExampleLowLevelRendering/CMakeFiles/ExampleLowLevelRendering.dir/Main.cpp.o  -o /home/constantine/Engines/BansheeEngine/bin/x64/Release/ExampleLowLevelRendering  -Wl,-rpath,/home/constantine/Engines/BansheeEngine/bin/x64/Release /home/constantine/Engines/BansheeEngine/bin/x64/Release/libBansheeEngine.so /home/constantine/Engines/BansheeEngine/bin/x64/Release/libBansheeCore.so /home/constantine/Engines/BansheeEngine/bin/x64/Release/libBansheeUtility.so /usr/lib/x86_64-linux-gnu/libSM.so /usr/lib/x86_64-linux-gnu/libICE.so /usr/lib/x86_64-linux-gnu/libX11.so /usr/lib/x86_64-linux-gnu/libXext.so /usr/lib/x86_64-linux-gnu/libXcursor.so /usr/lib/x86_64-linux-gnu/libXrandr.so /usr/lib/x86_64-linux-gnu/libXi.so && :
clang: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
/home/constantine/Engines/BansheeEngine/bin/x64/Release/libBansheeEngine.so: undefined reference to `std::vector<bs::ShaderVariation::Param, bs::StdAlloc<bs::ShaderVariation::Param, bs::GenAlloc> >::~vector()'
```

**Comment by procedural** [comment]
I'm no C++ expert, but it looks like a missing destructor somewhere? I may be wrong tho.

**Comment by BearishSun** [comment]
I believe it should be resolved with 7f236ff4701a57d36353b23751e021473877cbda. It seems to be some weird issue with Clang and the use of initializer lists on std::vector<T> within a shared library. Either a bug in Clang itself, or some obscure C++ spec. rule.

**Comment by procedural** [comment]
It compiles now, thanks!
